### PR TITLE
re-enable clickthrough

### LINKED ISSUE #111: worldmap linked variable not working - grafana 7
with the move to grafana 7.0 the click-through on the custom worldmap isn't working.

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #111 

The problem with the variable filter on the page remains - that may be resolved by https://github.com/grafana/grafana/pull/25192

For now though - this enables clicking on items on a map and that updating the linked variable, and for other panels to receive that update.

### CODE REVIEW

File: custom-plugins/grafana-worldmap-panel-msupply/src/worldmap.ts [hunk 1]
```diff
@@ -169,26 +169,29 @@ export default class WorldMap {
...
       fillOpacity: 0.5,
       location: dataPoint.key
     });
 
     this.createPopup(circle, dataPoint.locationName, dataPoint.valueRounded);
 
-    const ctrl = this.ctrl;
     circle.on('click', () => {
+      ctrl.variableSrv.init(ctrl.dashboard);
+      selectedFacility.options = [];
```
Comment (CarlosNZ) [inline custom-plugins/grafana-worldmap-panel-msupply/src/worldmap.ts:194]: Do we need this line? Not sure what it's doing, as the `.options` field has an empty array anyway.

Comment (mark-prins) [reply custom-plugins/grafana-worldmap-panel-msupply/src/worldmap.ts]: removed - with the addition of the `.init` call this isn't required.

File: custom-plugins/grafana-worldmap-panel-msupply/src/worldmap.ts [hunk 2]
```diff
@@ -169,26 +169,29 @@ export default class WorldMap {
   }
 
   createCircle(dataPoint) {
-    const selectedFacility = _.find(this.ctrl.vars, elem => {
-      return elem.name === this.ctrl.panel.linkedVariable;
+    const ctrl = this.ctrl;
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply custom-plugins/grafana-worldmap-panel-msupply/src/worldmap.ts]: added in 60c5e39

**Comment by mark-prins** [comment]
I'm not getting the filter-screws-up problem; just the filter-not-updated one, so perhaps that's a later grafana.
The fix mentioned above isn't in 7.0.4 so will have to wait for another release to test that.

**Comment by mark-prins** [comment]
> One quick thing -- can you increment the version number in `package.json` and the version/date in `src/plugin.json`? Cheers.

yep. good point, thanks.
Have tried the plugin with the pre-release version of grafana v7.1.0 and there's a problem - as the worldmap ctrl is injecting the `variableSrv` and this is no longer supported. Will remove and retest.

**Comment by CarlosNZ** [comment]
> Have tried the plugin with the pre-release version of grafana v7.1.0 and there's a problem - as the worldmap ctrl is injecting the `variableSrv` and this is no longer supported. Will remove and retest.

You used the `.getVarables()` method to get the selected facility. I wonder if we're now supposed to use the complementatry `replace` method to change it, as documented here: https://grafana.com/docs/grafana/latest/packages_api/runtime/templatesrv/

I tried looking for it when I was reviewing, but there was no `.replace()` method in `ctrl.dashboard` where the `getVariables()` method was, so maybe that's in 7.1?

**Comment by mark-prins** [comment]
> You used the `.getVariables()` method to get the selected facility. I wonder if we're now supposed to use the complementary `replace` method to change it, as documented here: https://grafana.com/docs/grafana/latest/packages_api/runtime/templatesrv/

I did start down that path - there is `replaceVariables()` available - but that is only useful for replacing the variable placeholder in a string with the current value of the variable. The `getVariables()` gave much the same result: you're right, it may not be kosher, but I think it's ok.

**Comment by mark-prins** [comment]
superceded by #161

**Pull request closed without merging**